instrumentation/aws-lambda-1.0/library/src/main/java/io/opentelemetry/instrumentation/awslambda/v1_0/ApiGatewayProxyRequest.java

```diff
@@ -42,9 +44,20 @@ static ApiGatewayProxyRequest forStream(final InputStream source) throws IOExcep
     return new CopiedApiGatewayProxyRequest(source);
   }
 
+  private static boolean nullOrEmpty(List<String> values) {
+    return ((values == null) || values.isEmpty());
+  }
+
+  private static String extractFirstValue(Map.Entry<String, List<String>> entry) {
+    List<String> values = entry.getValue();
+    return (nullOrEmpty(values) ? null : values.get(0));
```

Comment:
How about

```suggestion
    return entry.getValue().stream()
        .findFirst()
        .orElse(null);
```

instead?

Reply:
More readable, less performant I guess (streaming the list vs checking for empty()). Also, null values list is not handled properly.